drivers/osdp/src/osdp.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2020 Siddharth Chandrasekaran <siddharth@embedjournal.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <init.h>
+#include <device.h>
+#include <drivers/uart.h>
+#include <sys/ring_buffer.h>
+#include <logging/log.h>
+
+#include "osdp_common.h"
+
+LOG_MODULE_REGISTER(osdp, CONFIG_OSDP_LOG_LEVEL);
+
+struct osdp osdp_ctx;
+struct osdp_cp osdp_cp_ctx;
+struct osdp_pd osdp_pd_ctx[CONFIG_OSDP_NUM_CONNECTED_PD];
+
+static struct k_thread osdp_refresh_thread;
+static K_THREAD_STACK_DEFINE(osdp_thread_stack, 256);
```

Comment:
I recommend having a config option for stack size, it is way easier to change when the size needs to be changed.

Reply:
Ohh it was configurable, this instance didn't use it and was a bug. Thanks!